doc/extending/creating_a_numba_jax_op.rst

```diff
@@ -645,9 +645,15 @@ In out previous example of extending JAX, :class:`Eye`\ :class:`Op` was used wit
         # Pass the graph and any inputs to the testing function
         compare_jax_and_py(out_fg, [3])
 
-This one nowadays leads to a test failure due to new restrictions in JAX + JIT,
-as reported in issue `#654 <https://github.com/pymc-devs/pytensor/issues/654>`_.
-All jitted functions now must have constant shape, which means a graph like the
-one of :class:`Eye` can never be translated to JAX, since it's fundamentally a
-function with dynamic shapes. In other words, only PyTensor graphs with static shapes
-can be translated to JAX at the moment.
\ No newline at end of file
+cannot be translated to JAX, since it involved a function with dynamic shapes.
```

Comment:
The old non-example is still better. The new test is not even testing jax because the eye operations is constant folded before we call jax and jax never sees an eye

Reply:
I will make it a Note session about JAX unable to process graph with dynamic shape.

Reply:
Also note it's not that any dynamic shapes are disallowed. They can for instance depend on input shapes, so `eye(x.shape[0])` would probably work (or easily be made to work) even if x has a dynamic shape

Reply:
I included that.